api/cpp/include/slint_platform.h

```diff
@@ -227,6 +227,53 @@ class SoftwareRenderer
     }
 };
 
+/// An opaque, low-level window handle that internalizes everything necessary to exchange messages
+/// with the windowing system. This includes the connection to the display server, if necessary.
+///
+/// Note that this class does not provide any kind of ownership. The caller is responsible for
+/// ensuring that the pointers supplied to the constructor are valid throughout the lifetime of the
+/// WindowHandle.
+class WindowHandle
+{
+    cbindgen_private::CppRawHandleOpaque inner;
+
+public:
+    WindowHandle() = delete;
+#    if !defined(__APPLE__) && !defined(_WIN32) && !defined(_WIN64)
+    WindowHandle(uint32_t /*xcb_window_t*/ window, uint32_t /*xcb_visualid_t*/ visual_id,
+                 xcb_connection_t *connection, int screen)
+    {
+        cbindgen_private::slint_new_raw_window_handle_x11(window, visual_id, connection, screen,
+                                                          &inner);
+    }
+
+    WindowHandle(wl_surface *surface, wl_display *display)
+    {
+
+        cbindgen_private::slint_new_raw_window_handle_wayland(surface, display, size, &inner);
+    }
+
+#    elif defined(__APPLE__) && !defined(_WIN32) && !defined(_WIN64)
+
+    WindowHandle(void *nsview, void *nswindow)
+    {
+
+        cbindgen_private::slint_new_raw_window_handle_appkit(nsview, nswindow, &inner);
+    }
+
+#    elif !defined(__APPLE__) && (defined(_WIN32) || !defined(_WIN64))
+
+    /// Windows handle
+    WindowHandle(void *hwnd, void *hinstance)
```

Comment:
i wonder if we shouldn't have explicit  `from_win32`  and co. static function instead becuase these void* are not really typesafe.

Reply:
I entirely agree. I never liked the overloads to be honest. Will change.